### ISSUE TITLE
Unused path statement with unresolved references is highlighted as an error 

### ISSUE THREAD

**Issue by Undin** [intellij-rust/intellij-rust#9037]
This code triggers the false-positive, while compiling without issue:
```rust
enum Foo {
    Bar,
    Baz,
}

fn foo() {
    <Foo>::Bar;  // Unresolved reference: `Bar` 
}
```

_Originally posted by @wyfo in https://github.com/intellij-rust/intellij-rust/issues/8287#issuecomment-1165700755_
